djhtml/modes.py

```diff
@@ -151,6 +151,9 @@ def tokenize(self, tabwidth):
                 elif token:
                     line.append(token)
                     line_nr += token.text.count("\n")
+                    if token.recursive:
+                        # Recursive tokens report 1 more line than they should.
+                        line_nr -= 1
```

Comment:
While this indeed fixes the problem, I have a _strong_ suspicion the actual error is caused by this code on line 214:

        tokens.append(Token.Recursive(body + "\n", line_nr + 1, mode=DjNoOp))

Reply:
Correct, I think so too.
However: I tried changing that, and it caused changes of indenting. I experimented with more/less Newline tokens, removing the \n, and more, but couldn't get any solution to properly work except for this.

It's not just comments throwing off the count, but also style and script tags.

Reply:
(also: I don't 100% understand why that code adds both a \n to body, and a token.Newline, which seems odd)

Reply:
Yup, it's complicated. I'm considering doing multiple passes by the tokenizer instead of just one. It will be N times slower but the tokenization will be more consistent and hopefully more understandable.

For now however, this PR fixes the issue so I will happily accept it. Thanks!